### ISSUE TITLE
Astro CLI --> Deploy command with --pytest, -t and --test flag is not working in linux system

### ISSUE THREAD

**Issue by sanchit135** [astronomer/astro-cli#870]
**Describe the bug**

When in linux system below mentioned commands are executed 
./astro deploy --pytest --test tests -f
./astro deploy --pytest -t tests -f
./astro deploy --pytest -f

are not working as expected .

Expected Output:

dags should be deployed successfully with mentioned message

Successfully pushed Docker image to Astronomer registry.Navigate to the Astronomer UI for confirmation that your deploy was successful.

Actual Output :

Error: Something went wrong while Pytesting your local DAGs,
if the issue persists rerun the command without the '--pytest' flag to deploy: failed to execute cmd: exit status 5

```
Building image...
Sending build context to Docker daemon  46.03MB
Step 1/1 : FROM quay.io/astronomer/astro-runtime:6.0.3
6.0.3: Pulling from astronomer/astro-runtime
bd159e379b3b: Pull complete 
de08aeb7fd50: Pull complete 
6dd159998355: Pull complete 
4d872ce7a60f: Pull complete 
962145d175ca: Pull complete 
63cbde51f401: Pull complete 
92aa697a3158: Pull complete 
60c974dcaec9: Pull complete 
90c74a65f106: Pull complete 
6376873389d8: Pull complete 
ade3424e5989: Pull complete 
754bd5dc2106: Pull complete 
017cce4202b5: Pull complete 
3a73b0101a75: Pull complete 
177890fd4171: Pull complete 
b88ae19627ba: Pull complete 
14e424f31a21: Pull complete 
4ded246167bf: Pull complete 
ef5db2b0fe31: Pull complete 
b53c1cbfd226: Pull complete 
500475b930dc: Pull complete 
967e9cb12c1e: Pull complete 
a31772b2cc5a: Pull complete 
63acebc9a412: Pull complete 
0902632630a3: Pull complete 
309cab812caf: Pull complete 
4f4fb700ef54: Pull complete 
Digest: sha256:e8897a1720406d41a5cb6f721935191881b1560aa4ed323af37938c392af1d2b
Status: Downloaded newer image for quay.io/astronomer/astro-runtime:6.0.3
# Executing 7 build triggers
 ---> Running in aaeb92630b9a
Removing intermediate container aaeb92630b9a
 ---> Running in 7fc36b829334
+ [[ -s packages.txt ]]
Removing intermediate container 7fc36b829334
 ---> Running in 34489a8a10c1
+ grep -Eqx 'apache-airflow\s*[=~>]{1,2}.*' requirements.txt
+ pip install --no-cache-dir -r requirements.txt
Looking in indexes: https://pip.astronomer.io/v2/
Looking in links: https://pip.astronomer.io/simple/astronomer-fab-security-manager/, https://pip.astronomer.io/simple/astronomer-airflow-version-check/
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip available: 22.3 -> 22.3.1
[notice] To update, run: pip install --upgrade pip
Removing intermediate container 34489a8a10c1
 ---> Running in 4e959c4144d1
Removing intermediate container 4e959c4144d1
 ---> 36a33f2721b4
Successfully built 36a33f2721b4
Successfully tagged ecliptical-probe-7195/airflow:latest
Runtime Version: 6.0.3
testing ecliptical-probe-7195/airflow:latest
Testing image...
============================= test session starts ==============================
platform linux -- Python 3.9.15, pytest-7.1.3, pluggy-1.0.0
rootdir: /usr/local/airflow
plugins: anyio-3.6.2
collected 0 items

============================ no tests ran in 0.01s =============================
Error: Something went wrong while Pytesting your local DAGs,
if the issue persists rerun the command without the '--pytest' flag to deploy: failed to execute cmd: exit status 5
```

**What CLI Version did you experience this bug?**

**What Operating System is the above CLI installed on?**. Linux

**🪜 Steps To Reproduce**

**📸 Screenshots**

**Comment by kushalmalani** [comment]
@sanchit135 

1. What cli version are you on?
2. Did you change anything in your airflow directory (dags or test files)
3. Are you running the astro binary from inside the airflow project? If yes, can you try the same test by moving it to a different location?

**Comment by sanchit135** [comment]
1. I used cli version 1.7
2. I did not changed anything in airflow
3. I was using the astro binary inside the airflow project directory. I will try the other way also.

But these test are working as expected in mac, not working in linux

**Comment by sanchit135** [comment]
HI Team, When are planning to close this issue ?

**Comment by sunkickr** [comment]
It seems like the command is unable to find any pytest to run? are there any pytests located in the `./tests/` directory?

**Comment by kushalmalani** [comment]
Update: Seems this could be a instance user issue. When i exec into the built image and see the tests dags permissions i see

```
-rw-r--r-- 1 astro root 2385 Dec  5 18:32 test_dag_integrity.py
```

on my local machine, i can see that file having the permission

```
-rw-r--r-- 1 astro astro 2385 Nov 22 20:20 test_dag_integrity.py
```

**Comment by kushalmalani** [comment]
@sanchit135 This seems like how pip is installed on the ubuntu instance...when a docker image is built, i see the following

```+ pip install --no-cache-dir -r requirements.txt
Looking in indexes: https://pip.astronomer.io/v2/
Looking in links: https://pip.astronomer.io/simple/astronomer-fab-security-manager/, https://pip.astronomer.io/simple/astronomer-airflow-version-check/
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
```

**Comment by kushalmalani** [comment]
Was this working before and it broke recently? If yes, can you track me down as to what cli and runtime version we were using then?

I dont think this is a cli issue. Even if i do a docker built and use that image, its the same behavior...

**Comment by wolfier** [comment]
This issue is happening to a user running CLI version `1.16.2` and `1.17.1`.

The test file is present in the expected directory `.astro/test_dag_integrity_default.py` but pytest seems to be unable to find the file. The strange part is that it used to work in `1.16.2` but than suddenly stopped working after one commit that has nothing to do with tests or would invalidate the tests.

**Comment by kushalmalani** [comment]
This has been fixed in 1.19
